Algorithms/Problems/NQueenProblem/NQueenProblem.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Linq;
+
+namespace Algorithms.Problems
+{
+    public static class NQueenProblem
+    {
+        /// <summary>
+        /// Solves N-Queen Problem given a n dimension chessboard and using backtracking with recursion algorithm.
+        /// If we find a dead-end within or current solution we go back and try another position for queen.
+        /// </summary>
+        /// <param name="board">initial board.</param>
+        /// <param name="n">chessboard dimensions.</param>
+        /// <param name="col">starting column.</param>
+        /// <returns>Matrix solved. </returns>
+        public static bool BacktrackSolve(int[,] board, int n, int col)
+        {
+            if (n == 2 || n == 3)
+            {
+                // To return an error on known solution exceptions.
+                throw new ArgumentException("These dimensions do not have a solution.");
+            }
```

Comment:
Why error? There is no solution, but that's not an exceptional situation. false should be returned in this case or Enumerable.Empty<int[,]>() taking into account suggestions above.